wgsl/index.bs

```diff
@@ -988,6 +1031,16 @@ Restrictions on runtime-sized arrays:
     a store type in any other cases.
 * An expression must not evaluate to a runtime-sized array type.
 
+<pre class='def'>
+array_type_decl
+  | attribute_list* ARRAY LESS_THAN type_decl (COMMA element_count_expression)? GREATER_THAN
+
+element_count_expression
+  : INT_LITERAL
```

Comment:
should this be our constexpr thing?

Reply:
That depends on where #1272 goes. We can refactor rules later if it converges that way.